accessibility_api/AccessibilityRequestHandler.py

```diff
@@ -0,0 +1,129 @@
+'''
+This Source Code Form is subject to the terms of the Mozilla Public
+License, v. 2.0. If a copy of the MPL was not distributed with this file,
+You can obtain one at http://mozilla.org/MPL/2.0/.
+'''
+
+import sys
+from BaseHTTPServer import BaseHTTPRequestHandler
+from urlparse import urlparse, parse_qs, parse_qsl
+from json import dumps
+from comtypes import CoInitialize
+from accessibility_api.accessibility_lib.scripts.accessible import accessible
+from accessibility_api.accessibility_lib.scripts.event import event
+from accessibility_api.accessibility_lib.scripts.commands import (
+    execute_command
+)
+
+
+class AccessibilityRequestHandler(BaseHTTPRequestHandler):
+    """ Service requests for accessible objects """
+    def _successful_response(self, json):
+        self.send_response(200)
+        self.send_header('Content-Type', 'application/json')
+        self.end_headers()
+        self.wfile.write(json)
+
+    def _bad_request(self, message):
+        self.send_response(500, message)
+        self.send_header('Content-Type', 'application/json')
+        self.end_headers()
+
+    def _parse_identifiers(self, params):
+        name = params.get('name')
+        role = params.get('role')
+
+        identifiers = {}
+        if name is not None:
+            identifiers["Name"] = name
+        if role is not None:
+            identifiers["Role"] = role
+
+        return identifiers
+
+    def do_GET(self):
+        url = urlparse(self.path)
+        handler = getattr(self, "get_" + url.path[1:], None)
+        if handler:
+            handler(url.query)
+
+        else:
+            self.send_response(403, 'Invalid Request')
+            self.send_header('Content-Type', 'application/json')
+            self.end_headers()
+        return
+
+    def get_accessible(self, urlquery):
+        '''Retrieve accessible object'''
+
+        params = dict(parse_qsl(urlquery))
+        identifiers = self._parse_identifiers(params)
+        acc_obj = accessible(params.get('interface'), identifiers)
```

Comment:
I think we should avoid here and in any other function any processing of params. I think it would be much cleaner if we just pass params to accessible(), event() and execute_command(). Anything else should be done inside of those methods. I would also just do name->Name and role->Role on the stop, if it is needed.